File: 240708/문자에 따른 명령/text-based-commands.py
def get_new_position(commands):
    # 초기 위치와 방향 설정 (북쪽을 향함)
    x, y = 0, 0
    direction = 0  # 0: 북, 1: 동, 2: 남, 3: 서

    # 방향 벡터 설정 (북, 동, 남, 서 순)
    dx = [0, 1, 0, -1]
    dy = [1, 0, -1, 0]

    for command in commands:
        if command == 'L':
            direction = (direction - 1) % 4
        elif command == 'R':
            direction = (direction + 1) % 4
        elif command == 'F':
            x += dx[direction]
            y += dy[direction]

    return (x, y)

def different_positions(command_string):
    n = len(command_string)
    unique_positions = set()
    
    # 모든 가능한 잘못된 명령을 교정하여 도착 지점을 계산
    for i in range(n):
        for new_command in ['L', 'R', 'F']:
            if command_string[i] != new_command:
                new_commands = command_string[:i] + new_command + command_string[i+1:]
                final_position = get_new_position(new_commands)
                unique_positions.add(final_position)
    
    return len(unique_positions)

# 예제 사용
command_string = input().strip()
print(different_positions(command_string))

File: 240708/문자에 따른 명령/text-based-commands.cpp
#include <iostream>
#include <unordered_set>
#include <vector>
#include <string>

using namespace std;

// Function to simulate movement and return the final coordinates
pair<int, int> get_final_coordinates(const string& commands) {
    int direction = 0; // 0: north, 1: east, 2: south, 3: west
    int x = 0, y = 0;
    
    for (char command : commands) {
        if (command == 'L') {
            direction = (direction + 3) % 4;
        } else if (command == 'R') {
            direction = (direction + 1) % 4;
        } else if (command == 'F') {
            if (direction == 0) {
                y += 1;
            } else if (direction == 1) {
                x += 1;
            } else if (direction == 2) {
                y -= 1;
            } else if (direction == 3) {
                x -= 1;
            }
        }
    }
    
    return {x, y};
}

int find_distinct_destinations(const string& commands) {
    unordered_set<string> unique_points;
    vector<char> directions = {'L', 'R', 'F'};
    
    for (int i = 0; i < commands.size(); ++i) {
        for (char new_command : directions) {
            if (commands[i] == new_command) {
                continue;
            }
            
            string new_commands = commands;
            new_commands[i] = new_command;
            pair<int, int> final_coordinates = get_final_coordinates(new_commands);
            string point = to_string(final_coordinates.first) + "," + to_string(final_coordinates.second);
            unique_points.insert(point);
        }
    }
    
    return unique_points.size();
}

int main() {
    string commands;
    cin >> commands;
    
    cout << find_distinct_destinations(commands) << endl;
    
    return 0;
}